pinot-core/src/main/java/org/apache/pinot/core/segment/creator/JsonIndexCreator.java

```diff
@@ -0,0 +1,42 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.core.segment.creator;
+
+import java.io.Closeable;
+import java.io.IOException;
+
+
+/**
+ * Index creator for json index.
+ */
+public interface JsonIndexCreator extends Closeable {
```

Comment:
Do we not have an existing index creator interface (for example, text index creator could have used the same interface)?

Reply:
Not yet, we only have an interface for the inverted index. We can do the clean up later